vsintegration/src/FSharp.Editor/CodeFixes/FSharpCodeFixContext.fs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace Microsoft.VisualStudio.FSharp.Editor
+
+open Microsoft.CodeAnalysis
+open Microsoft.CodeAnalysis.Text
+
+open CancellableTasks
+
+type FSharpCodeFixContext(document: Document, span: TextSpan) =
```

Comment:
Ho often will this be allocated?

Reply:
I guess I can't make it a struct with all this async and lazy stuff?

Reply:
TBH this feels like it should just be a set of extensions to the regular context Roslyn gives us (e.g. extension methods)

Reply:
> TBH this feels like it should just be a set of extensions to the regular context Roslyn gives us (e.g. extension methods)

Yep, that would be much better

Reply:
@T-Gro @vzarytovskii 

So no, we cannot have these as Roslyn extensions for testability reasons. I mean, that's actually the whole point of having our own context for the core code fix logic - to make unit testing possible. We don't want to pass the real code fix context to the core logic of the code fix because we cannot really create/inherit one in tests.

What we can do is to have these as some kind of helpers. We already have `SymbolHelpers`, this is the same idea. This way:
1) We can have `FSharpCodeFixContext` as a struct
2) We make use of internal caching capabilities of Roslyn docs
3) We still remove some repetitive code

Keep in mind that we can iterate on this. More code fixes will be addressed soon - we'll see where this gets us.

Reply:
I must admit I do not get it.
The Roslyn's CodeFixContext is a `public readonly struct`, why is it not testable?

(even if the test code might need to write more boilerplate code to actually create it, that does not matter)

Reply:
Hah okay yes, after some exercise, I set things up in tests using the original context. Indeed, we don't need to build anything on top of it, so I over-engineered stuff here.

Now new shortcuts are done as extensions to the original Roslyn context.

Thanks for noticing this, I am a bit tunnel-visioned here so this perspective is helpful :)